### PR TITLE
Fixes #8794

### LINKED ISSUE #8794: Make it possible to use drag and drop in a webview
- VSCode Version: 1.3.0-insider
- OS Version: windows7

Steps to Reproduce:
1.  Download My extension:  https://github.com/f111fei/test-files/raw/master/webview-test-0.0.1.vsix
2.  Run command `Hello Webview` to open my webview.

Problem:
1.  Drag Button into TextArea. Dragfeedback overlay will prevent  my dragging action.
2.  Typing some text in the textarea. Use `Ctrl+A` to select all, Nothing happen.

Reason
1. see https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/electron-browser/window.ts#L91, the default drag handler is prevented.
2. see https://github.com/Microsoft/vscode/blob/master/src/vs/platform/keybinding/browser/keybindingServiceImpl.ts#L380, `selectAll` handler is prevented.

### PR DESCRIPTION
**Pull request by f111fei**
@bpasero  @jrieken   take a look

### CODE REVIEW

**Comment by jrieken** [comment]
We cannot hardcode keybindings like that. If someone defines `Ctrl+A` to let's say open file he is blocked by this. The commands that are bound by default to those keys should have a `when` condition which tells the keybindings service not run the command (and thereby consume the event).

File: src/vs/workbench/parts/html/browser/webview.ts [hunk 1]
```diff
@@ -45,6 +47,19 @@ KeybindingsRegistry.registerCommandDesc({
...
 type ApiThemeClassName = 'vscode-light' | 'vscode-dark' | 'vscode-high-contrast';
 
+function isDefaultKeyboardEvent(e: KeyboardEvent): boolean {
+	let keyEvent = new StandardKeyboardEvent(e);
+	let keybinding = keyEvent.asKeybinding();
+	switch (keybinding) {
+		case KeyMod.CtrlCmd | KeyCode.KEY_C:
+		case KeyMod.CtrlCmd | KeyCode.KEY_V:
+		case KeyMod.CtrlCmd | KeyCode.KEY_X:
+		case KeyMod.CtrlCmd | KeyCode.KEY_A:
```
Comment (jrieken) [inline src/vs/workbench/parts/html/browser/webview.ts]: this is bad

**Comment by f111fei** [comment]
@jrieken  selectAll actions maybe add a `when `condition `editorTextFocus`

**Comment by jrieken** [comment]
Yes, we have a similar issue with cut, copy, paste and now have a `when` condition

**Comment by f111fei** [comment]
@jrieken 	how about this? https://github.com/Microsoft/vscode/pull/8797/commits/77d061426e7bf981dcbe3c21ff677720f82dd542

File: src/vs/workbench/parts/html/browser/webview.ts [hunk 2]
```diff
@@ -93,6 +93,12 @@ export default class Webview {
 					this._webview.style.opacity = '';
 					return;
 				}
+			}),
+			addDisposableListener(this._webview, EventType.DRAG_OVER, (event: DragEvent) => {
+				event.stopImmediatePropagation();
+			}),
```
Comment (jrieken) [inline src/vs/workbench/parts/html/browser/webview.ts]: this leaks a listener

File: src/vs/workbench/parts/html/browser/webview.ts [hunk 3]
```diff
@@ -93,6 +93,12 @@ export default class Webview {
...
+			addDisposableListener(this._webview, EventType.DROP, (event: DragEvent) => {
+				event.stopImmediatePropagation();
 			})
```
Comment (jrieken) [inline src/vs/workbench/parts/html/browser/webview.ts]: same

File: src/vs/workbench/parts/html/browser/webview.ts [hunk 4]
```diff
@@ -93,6 +93,12 @@ export default class Webview {
 					this._webview.style.opacity = '';
 					return;
 				}
+			}),
+			addDisposableListener(this._webview, EventType.DRAG_OVER, (event: DragEvent) => {
+				event.stopImmediatePropagation();
+			}),
+			addDisposableListener(this._webview, EventType.DROP, (event: DragEvent) => {
+				event.stopImmediatePropagation();
```
Comment (jrieken) [inline src/vs/workbench/parts/html/browser/webview.ts]: same

**Comment by bpasero** [comment]
@jrieken @f111fei while the change seems to enable dropping stuff into the web view, I see issues:
* this disables any workbench related DND interaction if you go over the web view (e.g. you can no longer drop a file into the editor area to open the file)
* I am able to drop a URL into the web view and have it opened (see screen shot below)

Should we just be strict here and say that the HTML preview feature is not intended for DND operations?

image

**Comment by f111fei** [comment]
> this disables any workbench related DND interaction if you go over the web view (e.g. you can no longer drop a file into the editor area to open the file)

Workbench related DND interaction is contradictory with my usage. We can only choose one. 

> I am able to drop a URL into the web view and have it opened (see screen shot below)

Default if you drop a link into an electron window it will try to navigate to that URL. We maybe can prevent the interaction in my html extension.

**Comment by jrieken** [comment]
Closing as this needs further discussion/different changes

**Pull request closed without merging**